### PR TITLE
Add hostname and app to headers

### PR DESCRIPTION
**Pull request by secondhans**
Adding User-Agent field with both the hostname and a configurable identifier that is performing the call.

### CODE REVIEW

File: byterestclient/scriptinfo.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                "source": trc}
+
+    # from here on, we are in the interpreted case
+    scriptdir, trc = os.path.split(trc)
+    # if trc did not contain directory information,
+    # the current working directory is what we need
+    if not scriptdir:
+        scriptdir = os.getcwd()
+
+    scr_dict ={"name": trc,
```
Comment (genisd) [inline byterestclient/scriptinfo.py]: name and source appear to be the same. Not sure if that was intentional

Comment (sophiadavis) [inline byterestclient/scriptinfo.py]: Looks like it's intentional: there's a comment explaining in line 18.

Comment (secondhans) [inline byterestclient/scriptinfo.py]: They are when the script is not compiled. When it's a compiled python script, name will be the name of the compiled running code, while script will contain the name of the original script from which name was compiled.

**Comment by genisd** [comment]
that was the only thing I noticed.
so it's :+1:  from me :)

**Comment by allardhoeve** [comment]
That's a lot of source code :) What do you want to do with the header? Where will it show up? Apache or Nginx logs do not (currently) display this header in the logs.

Couldn't you have just added a `useragent=None` argument to `__init__` and passed it on to the remote using a standard header like `UserAgent`? That is much less code and achieves almost the same.

**Comment by allardhoeve** [comment]
This code is also probably full of nasty side-effects. For example, what does the code return when restclient is run inside of a gunicorn instance? Probably gunicorn.py or something. Or maybe not.

**Comment by allardhoeve** [comment]
:+1: when we remove the script thing and just add `myname=None` or `identifier=None` or even `useragent=None` or something.

File: byterestclient/__init__.py [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
 import requests
+import socket
 
 HTTPError = requests.HTTPError  # introspection
 
 
 class ByteRESTClient(object):
 
-    def __init__(self, token=None, endpoint=None):
+    def __init__(self, token=None, endpoint=None, identifier='byterestagent'):
```
Comment (allardhoeve) [inline byterestclient/__init__.py]: byterestclient?

**Comment by allardhoeve** [comment]
:+1:

**Pull request merged**